source/guides/first-steps/task-script.md

```diff
@@ -1,4 +1,121 @@
+```eval_rst
+.. contents:: Table of Contents
+    :local:
+```
+
+#### Task script basics
+
+A task script is a stand-alone script that can be run either via in-game command with the `/ex` command, as explained in [the guide to the /ex command](https://guide.denizenscript.com/guides/first-steps/ex-command.html), or by the `run` command in a script.
+
+Task scripts will run all of the Denizen commands that they include. A task script can be simple, it can be complicated, and it can even run other task scripts. This is useful to create logic-based script chains, like dialogue options or random content.
+
+#### Task script syntax
+
+Here's an example of a basic task script.
+
+```dscript_green
+example_task:
+    type: task
+    script:
+    - narrate "This is a basic task script!"
+```
+
+This script will narrate the text:
+
+```
+This is a basic task script!
+```
+
+to the player attached to the script. If you use the `/ex` command to run this script - specifically, by running `/ex run example_task` in-game, *you* will be the attached player.
+
+### Building your first task script in VS Code
+
+Previously, you've learned how to create an environment in VS Code, the editor that we recommend for writing Denizen scripts.
+
+#### Creating the file
+
+To create your first task script, start by opening your scripts folder in VS Code.
+
+TODO SCREENSHOT (open scripts folder)
+
+From there, right click the scripts folder in the explorer menu and click the "New File" option.
+
+TODO SCREENSHOT (new file highlighted)
+
+Once your file has been created, you can name it something like `MyTask.dsc`. Make sure to use the `.dsc` extension so VS Code recognizes it as a Denizen script, but the *name* of the file can be whatever you like. Usually, it's helpful to name the file something relevant to the script(s) that it contains.
+
+From here, double-click the file to open it in VS Code.
+
+TODO SCREENSHOT (opened file)
+
+Now you can begin writing your first task script!
+
+#### Writing the script
+
+Let's start with the core of the script:
+
+```dscript_blue
+my_first_task:
+    type: task
+    script:
+    - narrate (sometext)
+```
+
+This example should look familiar - it's very similar to the example above.
+
+### Script names
+
+The name of the script here is `myfirsttask`. It's at the top of the indentation. Each top-level indentation entry is a separate script. Here is an example of two different scripts:
+
+```dscript_green
+myfirsttask:
+    type: task
+    script:
+    - narrate "This is is task number one!"
+
+mysecondtask:
+    type: task
+    script:
+    - narratee "This is task number two!"
+```
+
+This example demonstrates what having two different scripts in the same file looks like. `myfirsttask` and `mysecondtask` are two different scripts, and each is a fully self-contained script.
```

Comment:
It might be good around here to explicitly emphasize that the _script names_ don't necessarily have anything to do with the _file name_ (since that tends to be a common confusion with beginners).